File: Selenium/Activities/Activity17.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import java.util.List;

public class Activity17 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/selects");
        System.out.println("Title of the page: " + driver.getTitle());
        // Store the select and make a new select object
        WebElement dropdown = driver.findElement(By.tagName("select"));
        Select select = new Select(dropdown);
        // Select the second option using the visible text
        select.selectByVisibleText("Option 2");
        System.out.println("Second option: " +select.getFirstSelectedOption().getText());
        // Select the third option using the index
        select.selectByIndex(3);
        System.out.println("Third option: " +select.getFirstSelectedOption().getText());
        // Select the fourth option using the value
        select.selectByValue("4");
        System.out.println("Fourth option: " +select.getFirstSelectedOption().getText());
        // Get all the options and print them
        List<WebElement> options = select.getOptions();
        System.out.println("All options: ");
        for(WebElement option : options){
            System.out.println(option.getText());
        }
        // Close the browser
        driver.quit();
    }
}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the browser to the URL
    driver.get("https://v1.training-support.net/selenium/selects")
    print("Title of the page is: ", driver.title)
    # Store the select and make a new select object
    dropdown = driver.find_element(By.TAG_NAME, "select")
    select = Select(dropdown)
    # Select the second option using the visible text
    select.select_by_visible_text("Option 2")
    print("Second option: ", select.first_selected_option.text)
    # Select the third option using the index
    select.select_by_index(3)
    print("Third option: ", select.first_selected_option.text)
    # Select the fourth option using the value
    select.select_by_value("4")
    print("Fourth option: ", select.first_selected_option.text)
    # Get all the options and print them
    options = select.options
    print("All options: ")
    for option in options:
        print(option.text)

